gammapy/scripts/image_pipe.py

```diff
@@ -11,242 +11,100 @@
+        events = events.select_energy(self.energy_band)
+        self.events = events.select_offset(self.offset_band)
 
+        self.maps = SkyMapCollection()
         self.empty_image = empty_image
         self.header = self.empty_image.to_image_hdu().header
         if exclusion_mask:
             self.maps['exclusion'] = exclusion_mask
-        self.ncounts_min = ncounts_min
 
-    def make_counts(self, obs_id):
-        """Fill the counts image for the events of one observation.
-
-        Parameters
-        ----------
-        obs_id : int
-            Observation ID
-        """
+        self.ncounts_min = ncounts_min
+        self.data_store = data_store
+        self.aeff = data_store.obs(self.obs_id).aeff
+        self.edisp = data_store.obs(self.obs_id).edisp
+        self.psf = data_store.obs(self.obs_id).psf
+        self.bkg = data_store.obs(self.obs_id).bkg
+        self.center = data_store.obs(self.obs_id).pointing_radec
+        self.livetime = data_store.obs(self.obs_id).observation_live_time_duration
+
+    def counts_map(self):
+        """Fill the counts image for the events of one observation."""
         counts_map = SkyMap.empty_like(self.empty_image)
-        obs = self.data_store.obs(obs_id=obs_id)
-        events = obs.events.select_energy(self.energy_band)
-        events = events.select_offset(self.offset_band)
-        counts_map.fill(value=events)
-        counts_map.data = counts_map.data
-
-        self.maps["counts"] = counts_map
-        return counts_map
-
-    def _low_counts(self, obs_id):
-        """Check if there are events for this observation after applying the offset and energy range.
-
-        Parameters
-        ----------
-        obs_id: int
-            Observation ID
-
-        Returns
-        -------
-        low_counts : bool
-            Does this observation contain low counts?
-        """
-        obs = self.data_store.obs(obs_id=obs_id)
-        events = obs.events.select_energy(self.energy_band)
-        events = events.select_offset(self.offset_band)
-
-        if len(events) <= self.ncounts_min:
-            log.warning('Skipping obs ' + str(obs_id) + ' because it only has ' + str(len(events)) + ' counts.')
-            return True
+        if len(self.events) > self.ncounts_min:
+            counts_map.fill(value=self.events)
         else:
-            return False
-
-    def make_total_counts(self):
-        """Stack the total count from the observation in the 'DataStore'.
-        """
-        total_counts = SkyMap.empty_like(self.empty_image)
-
-        for obs_id in self.obs_table['OBS_ID']:
-            if self._low_counts(obs_id):
-                continue
-            self.make_counts(obs_id)
-            total_counts.data += self.maps["counts"].data
-
-        self.maps["total_counts"] = total_counts
-        return total_counts
+            print("The total number of events {} is inferior to the requested minimal counts number".
```

Comment:
Can you change this to either use `log.warn('Too few counts...')`, when you want the program to continue or raise an error using `raise ValueError('Too few counts...')`, when it should stop.

The logging module can be used, when you put the following at the top of the file:

```
import logging
log = logging.getLogger(__name__)
```